File: clojure/p002.clj
; generate fib sequences up to 4 million, add up even values

(defn fib [a b]
  (if (< b 4000000)
    (+ (if (zero? (mod b 2))
         b 0)
       (fib b (+ a b)))
    0))

(println (fib 1 2))

File: go/p002.go
// generate fib sequence until 4 million, add up even values

package main

import "fmt"

const limit = 4000000

func main() {
	sum := 0
	for i,j := 1,2; j < limit; i,j = j,i+j {
		if j % 2 == 0 {
			sum += j
		}
	}
	fmt.Println(sum)
}